tensorflow/inference/docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -317,7 +323,7 @@ def _handle_invocation_post(self, req, res, model_name=None):
             if model_name:
                 if model_name not in self._mme_tfs_instances_status or \
                         not self._check_pid(self._mme_tfs_instances_status[model_name][0].pid):
-                    with lock():
+                    with lock:
```

Comment:
I think some of your locking is pretty course-grained, particularly around the whole loadmodel. Be aware that if an invocation comes in for a non-existent model during a load_model request, it will be blocked on acquiring the lock until the model is finished loading.

That's probably enough of a corner case that it's fine, though, I guess. I just wanted to point that out about the behavior for your awareness, at least.

Reply:
Yes, Thank you for point it out.